src/schema-convertors/internalToMongoDB.test.ts

```diff
@@ -0,0 +1,1768 @@
+      const internal = {
+        count: 2,
+        fields: [
+          {
+            name: 'mixedComplexType',
+            path: [
+              'mixedComplexType'
+            ],
+            count: 2,
+            type: [
+              'Array',
+              'Document',
+              'Undefined'
+            ],
+            probability: 0.6666666666666666,
+            hasDuplicates: false,
+            types: [
+              {
+                name: 'Array',
+                path: [
+                  'mixedComplexType'
+                ],
+                count: 1,
+                probability: 0.3333333333333333,
+                bsonType: 'Array',
+                types: [
+                  {
+                    name: 'Int32',
+                    path: [
+                      'mixedComplexType'
+                    ],
+                    count: 3,
+                    probability: 1,
+                    unique: 3,
+                    hasDuplicates: false,
+                    values: [
+                      1,
+                      2,
+                      3
+                    ],
+                    bsonType: 'Int32'
+                  }
+                ],
+                totalCount: 3,
+                lengths: [
+                  3
+                ],
+                averageLength: 3
+              },
+              {
+                name: 'Document',
+                path: [
+                  'mixedComplexType'
+                ],
+                count: 1,
+                probability: 0.3333333333333333,
+                bsonType: 'Document',
+                fields: [
+                  {
+                    name: 'a',
+                    path: [
+                      'mixedComplexType',
+                      'a'
+                    ],
+                    count: 1,
+                    type: 'String',
+                    probability: 1,
+                    hasDuplicates: false,
+                    types: [
+                      {
+                        name: 'String',
+                        path: [
+                          'mixedComplexType',
+                          'a'
+                        ],
+                        count: 1,
+                        probability: 1,
+                        unique: 1,
+                        hasDuplicates: false,
+                        values: [
+                          'bc'
+                        ],
+                        bsonType: 'String'
+                      }
+                    ]
+                  }
+                ]
+              },
+              {
+                name: 'Undefined',
+                bsonType: 'Undefined',
+                unique: 1,
+                hasDuplicates: false,
+                path: [
+                  'mixedComplexType'
+                ],
+                count: 1,
+                probability: 0.3333333333333333
+              }
+            ]
+          }
+        ]
+      };
```

Comment:
Not something for this PR, but do we plan to have integration tests at some point, e.g. ones that take documents, generate schemas for them, then add a validator to a real MongoDB database + try to insert those documents?